### ISSUE TITLE
UnhandledPromiseRejectionWarning: DBusError: Operation already in progress

### ISSUE THREAD

**Issue by alundstroem** [icanos/hassio-plejd#106]
Worked fine on HA prior to 116.0...

HassIO img on Intel Nuc 5.

`onDeviceConnected()
plejd-ble: warning: wasn't able to connect to Plejd, will retry.
plejd-ble: were unable to connect, will retry connection in 10 seconds.
(node:293) UnhandledPromiseRejectionWarning: DBusError: Operation already in progress
    at _methodReturnHandlers.(anonymous function) (/plejd/node_modules/dbus-next/lib/bus.js:339:27)
    at handleMessage (/plejd/node_modules/dbus-next/lib/bus.js:98:11)
    at EventEmitter.MessageBus.conn.on (/plejd/node_modules/dbus-next/lib/bus.js:147:9)
    at EventEmitter.emit (events.js:198:13)
    at /plejd/node_modules/dbus-next/lib/connection.js:112:14
    at Socket.<anonymous> (/plejd/node_modules/dbus-next/lib/message.js:63:9)
    at Socket.emit (events.js:198:13)
    at emitReadable_ (_stream_readable.js:555:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
(node:293) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:293) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
init()
plejd-ble: error: failed to start discovery. Make sure no other add-on is currently scanning. `

**Comment by alundstroem** [comment]
Hmm... a second reboot later and it works...

After first reboot bluetoothctl seemed fine, and dbus was recognized... dunno if this is a common bug?

I'll leave the issue open a while to see if there are any replies

**Comment by f00dagi** [comment]
Hi, totally unrelated, are you running the hassio image on a NUC and use bluetooth for Plejd?

All answers I have got from others is that the hassio image only supports the bluetooth device on raspberry pi. 

I have a NUC with built in bluetooth and have therefor an unsupported install with supervised home assistant on ubuntu but would love to get back to the hassio image.

Sorry for hijacking the thread, delete if inappropriate.

**Comment by alundstroem** [comment]
> Hi, totally unrelated, are you running the hassio image on a NUC and use bluetooth for Plejd?
> 
> All answers I have got from others is that the hassio image only supports the bluetooth device on raspberry pi.
> 
> I have a NUC with built in bluetooth and have therefor an unsupported install with supervised home assistant on ubuntu but would love to get back to the hassio image.
> 
> Sorry for hijacking the thread, delete if inappropriate.

@f00dagi I don't mind (: I first used a Raspberry Pi 3b+ (I think, Rpi3 anyways) and did a python venv install of HA... and used the older/first plejd custom_component (which required extracting some variables from the plejd app in your iPhone or Android device...) but then I wanted to move my installation to a "better" environment... and happened to have an Intel NUC 5 (i5) to spare...

Mind you... if I'd read _just a bit more_ than I did... I WOULD have done a "unsupported" installation as well... Instead I was choosing between using proxmox and running 2 virtual machines on my NUC, but my router ended up doing the work that I was targeting for the other machine, so I went on without proxmox...

T.M.I? (:

TL;DR; I'm currently on an Intel NUC 5 (i5) and a vanilla HassIO install from .img - and it works!

**Comment by f00dagi** [comment]
> 
> 
> > Hi, totally unrelated, are you running the hassio image on a NUC and use bluetooth for Plejd?
> > All answers I have got from others is that the hassio image only supports the bluetooth device on raspberry pi.
> > I have a NUC with built in bluetooth and have therefor an unsupported install with supervised home assistant on ubuntu but would love to get back to the hassio image.
> > Sorry for hijacking the thread, delete if inappropriate.
> 
> @f00dagi I don't mind (: I first used a Raspberry Pi 3b+ (I think, Rpi3 anyways) and did a python venv install of HA... and used the older/first plejd custom_component (which required extracting some variables from the plejd app in your iPhone or Android device...) but then I wanted to move my installation to a "better" environment... and happened to have an Intel NUC 5 (i5) to spare...
> 
> Mind you... if I'd read _just a bit more_ than I did... I WOULD have done a "unsupported" installation as well... Instead I was choosing between using proxmox and running 2 virtual machines on my NUC, but my router ended up doing the work that I was targeting for the other machine, so I went on without proxmox...
> 
> T.M.I? (:
> 
> TL;DR; I'm currently on an Intel NUC 5 (i5) and a vanilla HassIO install from .img - and it works!

**Comment by f00dagi** [comment]
> 
> 
> > Hi, totally unrelated, are you running the hassio image on a NUC and use bluetooth for Plejd?
> > All answers I have got from others is that the hassio image only supports the bluetooth device on raspberry pi.
> > I have a NUC with built in bluetooth and have therefor an unsupported install with supervised home assistant on ubuntu but would love to get back to the hassio image.
> > Sorry for hijacking the thread, delete if inappropriate.
> 
> @f00dagi I don't mind (: I first used a Raspberry Pi 3b+ (I think, Rpi3 anyways) and did a python venv install of HA... and used the older/first plejd custom_component (which required extracting some variables from the plejd app in your iPhone or Android device...) but then I wanted to move my installation to a "better" environment... and happened to have an Intel NUC 5 (i5) to spare...
> 
> Mind you... if I'd read _just a bit more_ than I did... I WOULD have done a "unsupported" installation as well... Instead I was choosing between using proxmox and running 2 virtual machines on my NUC, but my router ended up doing the work that I was targeting for the other machine, so I went on without proxmox...
> 
> T.M.I? (:
> 
> TL;DR; I'm currently on an Intel NUC 5 (i5) and a vanilla HassIO install from .img - and it works!

I went through the same evolution, rpi3 > rpi4 > nuc and also started with the first custom_component. I actually tried the HassIO vanilla installed on my NUC but didnt get it working. Now, with your success in mind I probably have to go back...

**Comment by alundstroem** [comment]
> > > Hi, totally unrelated, are you running the hassio image on a NUC and use bluetooth for Plejd?
> > > All answers I have got from others is that the hassio image only supports the bluetooth device on raspberry pi.
> > > I have a NUC with built in bluetooth and have therefor an unsupported install with supervised home assistant on ubuntu but would love to get back to the hassio image.
> > > Sorry for hijacking the thread, delete if inappropriate.
> > 
> > 
> > @f00dagi I don't mind (: I first used a Raspberry Pi 3b+ (I think, Rpi3 anyways) and did a python venv install of HA... and used the older/first plejd custom_component (which required extracting some variables from the plejd app in your iPhone or Android device...) but then I wanted to move my installation to a "better" environment... and happened to have an Intel NUC 5 (i5) to spare...
> > Mind you... if I'd read _just a bit more_ than I did... I WOULD have done a "unsupported" installation as well... Instead I was choosing between using proxmox and running 2 virtual machines on my NUC, but my router ended up doing the work that I was targeting for the other machine, so I went on without proxmox...
> > T.M.I? (:
> > TL;DR; I'm currently on an Intel NUC 5 (i5) and a vanilla HassIO install from .img - and it works!
> 
> I went through the same evolution, rpi3 > rpi4 > nuc and also started with the first custom_component. I actually tried the HassIO vanilla installed on my NUC but didnt get it working with bluetooth. Now, with your success in mind I probably have to go back...

Yeah, depending on what version of NUC you have, of course... but I've had zero problems up until this... but a reboot resolved that...

**Comment by Mikro007** [comment]
`(node:294) UnhandledPromiseRejectionWarning: DBusError: Failed to activate service 'org.bluez': timed out (service_start_timeout=25000ms)
    at _methodReturnHandlers.(anonymous function) (/plejd/node_modules/dbus-next/lib/bus.js:339:27)
    at handleMessage (/plejd/node_modules/dbus-next/lib/bus.js:98:11)
    at EventEmitter.MessageBus.conn.on (/plejd/node_modules/dbus-next/lib/bus.js:147:9)
    at EventEmitter.emit (events.js:198:13)
    at /plejd/node_modules/dbus-next/lib/connection.js:112:14
    at Socket.<anonymous> (/plejd/node_modules/dbus-next/lib/message.js:63:9)
    at Socket.emit (events.js:198:13)
    at emitReadable_ (_stream_readable.js:555:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
(node:294) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:294) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.`

I got this on my Rpi 3 b+???

**Comment by icanos** [comment]
Please reopen if this still is an issue.
